### PR TITLE
automatically phpunit in any directory containing phpunit.xml

### PR DESCRIPTION
**Pull request by bshaffer**
@tmatsuo PTAL

we were missing `appengine/standard/twilio` and `appengine/standard/mail` tests

### CODE REVIEW

**Comment by bshaffer** [comment]
If you want to explicitly include the `DIRS`, we could add something like this at the bottom to check all tests are run:

```bash
# make sure we didn't miss any tests
array_contains () {
    local seeking=$1; shift
    local in=1
    for element; do
        if [[ $element == $seeking ]]; then
            in=0
            break
        fi
    done
    return $in
}

ALL_TESTS_RUN="true"
# change to root dir (for find command)
cd $( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )/..
find * -name phpunit.xml -not -path '*/vendor/*' -exec dirname {} \; | while read line
do
    if ! array_contains "$line" "${DIRS[@]}"; then
      echo "ERROR - no test for ${line}"
      ALL_TESTS_RUN="false"
    fi
done

if [ "${ALL_TESTS_RUN}" = "false" ]; then
    exit 1
fi
```

File: testing/run_test_suite.sh [hunk 1]
```diff
@@ -20,42 +20,22 @@ if [ "${RUN_CS_FIXER}" = "true" ]; then
...
-  fi
-  popd
+# loop through all directories containing "phpunit.xml" and run them
+find * -name phpunit.xml -not -path '*/vendor/*' -exec dirname {} \; | while read DIR
+do
+    pushd ${DIR}
+    if [ -f "composer.json" ]; then
+        composer install
+    fi
+    if [ "$DIR" -eq "appengine/standard/modules" ]; then
```
Comment (tmatsuo) [inline testing/run_test_suite.sh]: Little bit concern on this, but maybe I will change the test in `appengine/standard/modules` to not needing this special care.

**Comment by tmatsuo** [comment]
LGTM

Thanks!
Before merging, please check the coveralls site and see whether you have expected files are listed there.

File: testing/run_test_suite.sh (hunk 1 above)
Comment (tmatsuo) [inline testing/run_test_suite.sh]: maybe you can just use
```bash
if [ "$DIR" = "appengine/standard/modules" ]; then
```
?

Comment (bshaffer) [inline testing/run_test_suite.sh]: @tmatsuo looks like we can add default env-vars to `phpunit.xml` [like this](http://elnur.pro/using-environment-variables-to-add-flexibility-to-phpunit-tests/). I like this approach

Comment (bshaffer) [inline testing/run_test_suite.sh]: @tmatsuo I see you purposefully defaulted it to just `app.yaml`, so maybe using `phpunit.xml` is not desirable?

Comment (tmatsuo) [inline testing/run_test_suite.sh]: Yeah, please try adding the envvar to the `phpunit.xml`!

Comment (tmatsuo) [inline testing/run_test_suite.sh]: No problem at all with using `phpunit.xml`.

Comment (tmatsuo) [inline testing/run_test_suite.sh]: or maybe I can do that in the next cl

**Comment by bshaffer** [comment]
@tmatsuo it looks [good to me](https://coveralls.io/builds/5945336), but I'm not sure how to see the individual files that were captured.

**Comment by tmatsuo** [comment]
LGTM, let's merge it

**Pull request merged**